### PR TITLE
[RNMobile] Activate preformat on android

### PR DESCRIPTION
**Pull request by marecar3**
## Description
Activate `pre` format block on Android platform: https://github.com/wordpress-mobile/gutenberg-mobile/issues/1264

GB mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1615

## How has this been tested?
Follow test instructions on `gb-mobile` PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/1615

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR. .

### CODE REVIEW

**Comment by marecar3** [comment]
Changed target branch to `master` as we don't want to hurry with this feature (it still has some active issues)

**Comment by hypest** [comment]
> Changed target branch to master

I think this PR also needs rebasing on top of master to only include its own commits. E.g. the "Spacer" commit is not really part of the PR.

Screenshot 2019-11-28 at 15 25 24

**Comment by marecar3** [comment]
> > Changed target branch to master
> 
> I think this PR also needs rebasing on top of master to only include its own commits. E.g. the "Spacer" commit is not really part of the PR.
> 
> Screenshot 2019-11-28 at 15 25 24

Thanks, @hypest  I have fixed that one.

File: packages/rich-text/src/component/index.native.js [hunk 1]
```diff
@@ -586,6 +586,15 @@ export class RichText extends Component {
 	}
 
 	willTrimSpaces( html ) {
+		const {
+			tagName,
+		} = this.props;
+
+		// aztec won't trimm spaces in a case of <pre> block, so we are excluding it
```
Comment (hypest) [inline packages/rich-text/src/component/index.native.js]: Nit: typo in `trimm`, let's replace with `trim`.

File: packages/rich-text/src/component/index.native.js [hunk 2]
```diff
@@ -586,6 +586,15 @@ export class RichText extends Component {
...
+		if ( tagName === 'pre' ) {
```
Comment (hypest) [inline packages/rich-text/src/component/index.native.js]: Nice and clear way of shortcircuiting `willTrimSpaces` in the Preformatted block case!

Can you also add a test in https://github.com/WordPress/gutenberg/blob/master/packages/rich-text/src/component/test/index.native.js to cover that case? Thanks!

**Review by hypest — COMMENTED** [review]
This works nicely! Let's add the unit test for the new `willTrimSpaces` codepath and fix the minor typo and I can review again!

File: packages/rich-text/src/component/index.native.js (hunk 2 above)
Comment (marecar3) [reply packages/rich-text/src/component/index.native.js]: Good suggestion, thanks!

**Comment by marecar3** [comment]
> This works nicely! Let's add the unit test for the new `willTrimSpaces` codepath and fix the minor typo and I can review again!

Hey @hypest, I have added unit test and also I have fixed minor typo. 
You can do another iteration of review :)

**Comment by hypest** [comment]
👋 @marecar3 , seems that the PR needs updating from master and there are a couple of lint issues to fix too.

**Comment by marecar3** [comment]
> 👋 @marecar3 , seems that the PR needs updating from master and there are a couple of lint issues to fix too.

👋 @hypest , I have update this PR with the latest master and I am not sure about lint issues, as commit before this one was with ✅ https://github.com/WordPress/gutenberg/runs/336677088
but maybe I am missing something.

**Comment by hypest** [comment]
Sorry, I think I didn't notice [the commit that fixed them before](https://github.com/WordPress/gutenberg/pull/18777/commits/3b4adb5f1831d0978685b1ee01c9fe761509425f)! Sorry for the confusion 😞

**Review by hypest — APPROVED** [review]
LGTM!

**Review by SergioEstevao — APPROVED** [review]
:shipit:

**Pull request merged**